apps/homescreen/test/unit/metadata_test.js

```diff
@@ -0,0 +1,205 @@
+/* global MockIndexedDB, HomeMetadata */
+'use strict';
+
+require('/shared/test/unit/mocks/mock_indexedDB.js');
+require('/js/metadata.js');
+
+suite('HomeMetadata', () => {
+  var mockIndexedDB;
+  var metadata;
+
+  setup(() => {
+    mockIndexedDB = new MockIndexedDB();
+    metadata = new HomeMetadata();
+  });
+
+  teardown(() => {
+    metadata = null;
+    mockIndexedDB.mTearDown();
+  });
+
+  suite('HomeMetadata#init()', () => {
+    test('should open an indexedDB instance', done => {
+      assert.equal(mockIndexedDB.dbs.length, 0);
+      metadata.init().then(() => {
+        assert.equal(mockIndexedDB.dbs.length, 1);
+        done();
+      });
+    });
+  });
+
+  suite('HomeMetadata#upgradeSchema()', () => {
+    var names = [];
+    var onUpgradeNeededEvent = version => {
+      return {
+        target: {
+          result: {
+            createObjectStore: (name, options) => {
+              names.push(name);
+              return {
+                createIndex: (name, keyPath, options) => {
+                }
+              };
```

Comment:
Same nit wrt the brackets here - this read read more easily as `return { createIndex: (name, key, options) => {} };`